contracts/Plasma.sol

```diff
@@ -0,0 +1,194 @@
+pragma solidity 0.4.19;
+
+import 'zeppelin-solidity/contracts/ownership/Ownable.sol';
+import 'zeppelin-solidity/contracts/math/SafeMath.sol';
+import 'zeppelin-solidity/contracts/math/Math.sol';
+import './lib/MerkleProof.sol';
+import './DataStructures/PriorityQueue.sol';
+import './Transaction.sol';
+
+import "./lib/RLP.sol";
+
+contract Plasma is Ownable {
+
+    mapping(address => bool) public operators;
+
+    modifier onlyOperator() {
+        if (operators[msg.sender]) _;
+    }
+
+    function setOperator(address _op, bool _status) public returns (bool success) {
+        require(msg.sender == owner);
+        operators[_op] = _status;
+        return true;
+    }
+
+    function isOperator() public view returns (bool success) {
+        return operators[msg.sender];
+    }
+
+    // blocks
+    struct Block {
+        uint blockIndex;
+        bytes32 merkleRoot;
+        uint256 timestamp;
+    }
+
+    uint public blockCount;
+    mapping(uint => Block) public childChain;
+
+    event BlockSubmitted(uint indexed _blockIndex, address _operator, bytes32 indexed _merkleRoot);
+
+    function submitBlock(bytes32 _merkleRoot, uint _blockIndex) public onlyOperator {
+        require(blockCount + 1 == _blockIndex);
+
+        Block memory newBlock = Block({
+            blockIndex: _blockIndex,
+            merkleRoot: _merkleRoot,
+            timestamp: block.timestamp
+        });
+
+        blockCount = _blockIndex;
+
+        childChain[blockCount] = newBlock;
+
+        BlockSubmitted(_blockIndex, msg.sender, _merkleRoot);
+    }
+
+    // deposits
+
+    mapping(uint => bytes32) public depositIndex;
+    mapping(bytes32 => uint) public depositBalance;
+    uint public depositCount;
+
+    event Deposit(uint indexed _depositIndex, bytes32 indexed _uid, address _depositor, uint _amount);
+
+    function deposit(address _currency, uint _amount) payable public {
+        require(_amount > 0);
+
+        if (_currency == address(0)) {
+            require(_amount == msg.value);
+        } else {
+            throw;
+        }
+        bytes32 uid = keccak256(msg.sender, _currency, depositCount);
```

Comment:
Any specific reason why you're using a hash as the token ID instead of `depositCount`?

Reply:
no reason. keccak256(depositCount) must be unpredictable. Thanks!

Reply:
keccak256(msg.sender, _currency, depositCount) restricts the call to msg.sender. There's no reason beside that. Changed to depositCount. Thanks.